iam/role/role.go

```diff
@@ -0,0 +1,114 @@
+	Tags                      map[string]string      `json:"tags"`
+	RoleName                  string                 `json:"roleName"`
+	TrustedArn                string                 `json:"trustedArn"`
+	Policy                    map[string]interface{} `json:"policy"`
+	MaxSessionDurationMinutes int                    `json:"maxSessionDurationMinutes"`
+	Path                      string                 `json:"path"`
+	ExternalIds               string                 `json:"externalIds"`
+}
+
+func (rsp *IAMRoleStackProps) setDefaults() {
+	if rsp.RoleName == "" {
+		rsp.RoleName = os.Getenv("ACORN_EXTERNAL_ID")
+	}
+	if rsp.MaxSessionDurationMinutes == 0 {
+		rsp.MaxSessionDurationMinutes = 60
+	}
+	if rsp.Path == "" {
+		rsp.Path = "/"
+	}
+}
+
+func (rsp *IAMRoleStackProps) validateProps() error {
+	var errs []error
+	if rsp.MaxSessionDurationMinutes < 60 {
+		errs = append(errs, fmt.Errorf("maxSessionDurationMinutes must be at least 60"))
+	}
+	if !strings.HasPrefix(rsp.Path, "/") {
+		errs = append(errs, fmt.Errorf("path must start with a /"))
+	}
+	if len(rsp.Policy) == 0 {
+		errs = append(errs, fmt.Errorf("policy cannot be empty"))
+	}
+	if _, err := arn.Parse(rsp.TrustedArn); err != nil {
+		errs = append(errs, fmt.Errorf("failed to parse trustedArn: %w", err))
+	}
+	return errors.Join(errs...)
+}
+
+func newIAMRoleStack(scope constructs.Construct, id string, props *IAMRoleStackProps) (awscdk.Stack, error) {
+	var sprops awscdk.StackProps
+	if props != nil {
+		sprops = props.StackProps
+	}
+	stack := awscdk.NewStack(scope, &id, &sprops)
+
+	roleProps := &awsiam.RoleProps{
+		AssumedBy:          awsiam.NewArnPrincipal(jsii.String(props.TrustedArn)),
+		Description:        jsii.String("Acorn created IAM Role"),
+		InlinePolicies:     &map[string]awsiam.PolicyDocument{"inline": awsiam.PolicyDocument_FromJson(props.Policy)},
+		MaxSessionDuration: awscdk.Duration_Minutes(jsii.Number(props.MaxSessionDurationMinutes)),
+		Path:               jsii.String(props.Path),
+		RoleName:           jsii.String(props.RoleName),
+	}
+	if props.ExternalIds != "" {
+		externalIds := make([]*string, len(strings.Split(props.ExternalIds, ",")))
+		for i, v := range strings.Split(props.ExternalIds, ",") {
+			externalIds[i] = jsii.String(v)
+		}
+
+		roleProps.ExternalIds = &externalIds
+	}
+
+	iamRole := awsiam.NewRole(stack, jsii.String(props.RoleName), roleProps)
+
+	awscdk.NewCfnOutput(stack, jsii.String("IAMRoleArn"), &awscdk.CfnOutputProps{
+		Value: iamRole.RoleArn(),
+	})
+
+	return stack, nil
+}
+
+func main() {
+	defer jsii.Close()
+
+	app := common.NewAcornTaggedApp(nil)
+	stackProps := &IAMRoleStackProps{
+		StackProps: *common.NewAWSCDKStackProps(),
+	}
+
+	if err := common.NewConfig(stackProps); err != nil {
+		logrus.Fatal(err)
+	}
+	stackProps.setDefaults()
+	if err := stackProps.validateProps(); err != nil {
+		logrus.Fatalf("invalid stack properties: %w", err)
+	}
+
+	common.AppendScopedTags(app, stackProps.Tags)
```

Comment:
Can we add a tag the indicates that this is created by acorn?

Reply:
Tags are already added by the common library. Here is an example:

![image](https://github.com/acorn-io/aws/assets/53102776/b7fedbe1-785d-4100-bc37-0e35af716e43)

Reply:
Oh sweet, very nice.